akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -0,0 +1,256 @@
+    import akka.http.scaladsl.model.HttpMethods._
+    import akka.http.scaladsl.model._
+
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorFlowMaterializer()
+
+    val requestHandler: HttpRequest => HttpResponse = {
+      case HttpRequest(GET, Uri.Path("/"), _, _, _) =>
+        HttpResponse(entity = HttpEntity(MediaTypes.`text/html`,
+          "<html><body>Hello world!</body></html>"))
+
+      case HttpRequest(GET, Uri.Path("/ping"), _, _, _) =>
+        HttpResponse(entity = "PONG!")
+
+      case HttpRequest(GET, Uri.Path("/crash"), _, _, _) =>
+        sys.error("BOOM!")
+
+      case _: HttpRequest =>
+        HttpResponse(404, entity = "Unknown resource!")
+    }
+
+    Http().bindAndHandleSync(requestHandler, "localhost", 8080)
+  }
+
+  // format: OFF
+
+  "high-level-server-example" in {
+    import akka.stream.ActorFlowMaterializer
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.server.Directives._
+    import akka.http.scaladsl.marshallers.xml.ScalaXmlSupport._
+
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorFlowMaterializer()
+
+    val route =
+      get {
+        pathSingleSlash {
+          complete {
+            <html>
+              <body>Hello world!</body>
+            </html>
+          }
+        } ~
+        path("ping") {
+          complete("PONG!")
+        } ~
+        path("crash") {
+          sys.error("BOOM!")
+        }
+      }
+
+    Http().bindAndHandle(route, "localhost", 8080)
+  }
+
+  "minimal-routing-example" in {
+    import akka.stream.ActorFlowMaterializer
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.server.Directives._
+    import akka.http.scaladsl.marshallers.xml.ScalaXmlSupport._
+
+    object Main extends App {
+      implicit val system = ActorSystem("my-system")
+      implicit val materializer = ActorFlowMaterializer()
+
+      val route =
+        path("hello") {
+          get {
+            complete {
+              <h1>Say hello to spray</h1>
+            }
+          }
+        }
+
+      val bindingFuture = Http().bindAndHandle(route, "localhost", 8080)
+
+      println(s"Server online at http://localhost:8080/\nPress RETURN to stop...")
+      Console.readLine()
+
+      import system.dispatcher // for the future transformations
+      bindingFuture
+        .flatMap(_.unbind()) // trigger unbinding from the port
+        .onComplete(_ ⇒ system.shutdown()) // and shutdown when done
+    }
+  }
+
+  "long-routing-example" in {
+    import akka.actor.ActorRef
+    import akka.util.Timeout
+    import akka.pattern.ask
+    import akka.http.scaladsl.marshalling.ToResponseMarshaller
+    import akka.http.scaladsl.unmarshalling.FromRequestUnmarshaller
+    import akka.http.scaladsl.model.StatusCodes.MovedPermanently
+    import akka.http.scaladsl.coding.Deflate
+    import akka.http.scaladsl.server.Directives._
+
+    // types used by the API routes
+    type Money = Double
```

Comment:
Money as floating point? :) Personally I think it is fine for an example, but I would bet that there will be users pointing out this.